spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskSchedulerController.java

```diff
@@ -0,0 +1,159 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller for operations on {@link ScheduleInfo}. This includes CRUD operations.
+ *
+ * @author Glenn Renfro
+ */
+@RestController
+@RequestMapping("/tasks/schedules")
+@ExposesResourceFor(ScheduleInfoResource.class)
+public class TaskSchedulerController {
+
+	private SchedulerService schedulerService;
+	private final Assembler taskAssembler = new Assembler();
+
+	/**
+	 * Create a {@code TaskSchedulerController} that delegates
+	 * <ul>
+	 * <li>task scheduling operations via {@link SchedulerService}</li>
+	 * </ul>
+	 *
+	 * @param schedulerService the scheduler service to use to delegate task
+	 *                            scheduler operations.
+	 */
+	public TaskSchedulerController(SchedulerService schedulerService) {
+		Assert.notNull(schedulerService, "schedulerService must not be null");
+		this.schedulerService = schedulerService;
+	}
+
+	/**
+	 * Return a page-able list of {@link ScheduleInfo}s.
+	 *
+	 * @param pageable  page-able collection of {@code ScheduleResource}.
+	 * @param assembler assembler for the {@link ScheduleInfo}
+	 * @return a list of Schedules
+	 */
+	@RequestMapping(value = "", method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public PagedResources<ScheduleInfoResource> list(Pageable pageable,
+			PagedResourcesAssembler<ScheduleInfo> assembler) {
+
+		List<ScheduleInfo> result = this.schedulerService.list(pageable);
+		Page<ScheduleInfo> page = new PageImpl<>(result, pageable,
+				result.size());
+		return assembler.toResource(page, taskAssembler);
+	}
+
+	/**
+	 * Return a page-able list of {@link ScheduleInfo}s for a specific
+	 * {@link org.springframework.cloud.dataflow.core.TaskDefinition} name.
+	 *
+	 * @param pageable page-able collection of {@code ScheduleResource}.
+	 * @param assembler assembler for the {@link ScheduleInfo}
+	 * @param taskDefinitionName name of the taskDefinition to search.
+	 * @return a list of Schedules
+	 */
+	@RequestMapping("/instances/{taskDefinitionName}")
+	public PagedResources<ScheduleInfoResource> filteredList(Pageable pageable, @PathVariable String taskDefinitionName,
+			PagedResourcesAssembler<ScheduleInfo> assembler) {
+		List<ScheduleInfo> result = this.schedulerService.list(pageable, taskDefinitionName);
+		Page<ScheduleInfo> page = new PageImpl<>(result, pageable,
+				result.size());
+		return assembler.toResource(page, taskAssembler);
+	}
+
+	/**
+	 * Create a schedule for an existing
+	 * {@link org.springframework.cloud.dataflow.core.TaskDefinition} name.
+	 *
+	 * @param scheduleName the name of the schedule being created.
+	 * @param taskDefinitionName the name of the existing task to be executed (required)
+	 * @param properties the runtime properties for the task, as a comma-delimited list of
+	 * key=value pairs
+	 * @param arguments the runtime commandline arguments
+	 * @return instance of the schedule
```

Comment:
looks like we don't return the instance based on the contract.

Reply:
done